synctool/src/main/java/org/duracloud/sync/walker/DirWalker.java

```diff
@@ -39,42 +35,17 @@
     private boolean continueWalk;
 
     private List<File> filesAndDirs;
-    private WildcardFileFilter fileFilter;
-    private ChangedList fileList;
+ 
+    protected final ChangedList changedList;
     private int files = 0;
     private boolean complete = false;
 
-    protected DirWalker(List<File> filesAndDirs, File excludeFile) {
+    protected DirWalker(List<File> filesAndDirs) {
         super();
         this.filesAndDirs = filesAndDirs;
-        fileList = ChangedList.getInstance();
-
-        if(null != excludeFile) {
```

Comment:
Without an exclude check here, there's the potential of processing a large number of irrelevant files (in an excluded directory.)